driver/sql/projection_notification_processor.go

```diff
@@ -4,33 +4,38 @@ import (
 	"context"
 	"runtime"
 	"sync"
+	"time"
 
 	"github.com/hellofresh/goengine"
 	"github.com/pkg/errors"
 )
 
-// Ensure the projectionNotificationProcessor.Queue is a ProjectionTrigger
-var _ ProjectionTrigger = (&projectionNotificationProcessor{}).Queue
-
 type (
-	// projectionNotificationProcessor provides a way to Trigger a notification using a set of background processes.
-	projectionNotificationProcessor struct {
+	// ProjectionNotificationProcessor provides a way to Trigger a notification using a set of background processes.
+	ProjectionNotificationProcessor struct {
 		done            chan struct{}
-		queue           chan *ProjectionNotification
 		queueProcessors int
-		queueBuffer     int
 
 		logger  goengine.Logger
 		metrics Metrics
+
+		notificationQueue NotificationQueueInterface
 	}
 
 	// ProcessHandler is a func used to trigger a notification but with the addition of providing a Trigger func so
 	// the original notification can trigger other notifications
 	ProcessHandler func(context.Context, *ProjectionNotification, ProjectionTrigger) error
 )
 
-// newBackgroundProcessor create a new projectionNotificationProcessor
-func newBackgroundProcessor(queueProcessors, queueBuffer int, logger goengine.Logger, metrics Metrics) (*projectionNotificationProcessor, error) {
+// NewBackgroundProcessor create a new projectionNotificationProcessor
+func NewBackgroundProcessor(
+	queueProcessors,
+	queueBuffer int,
+	logger goengine.Logger,
+	metrics Metrics,
+	retryDelay time.Duration,
```

Comment:
If we allow passing in `NotificationQueueInterface` do we really need the `retryDelay`?